### PR TITLE
Implemented packaging using CPack

### PR DESCRIPTION
**Pull request by KOLANICH**

### CODE REVIEW

**Comment by lemire** [comment]
This PR breaks the build under Visual Studio 2017, see CI tests...

```
LINK : fatal error LNK1104: cannot open file 'Debug\FastPFOR.lib' [C:\projects\fastpfor\build\FastPFOR_unittest.vcxproj]
```

File: CMakeLists.txt [hunk 1]
```diff
@@ -5,6 +5,7 @@
 #
 cmake_minimum_required(VERSION 2.8.7)
 set (CMAKE_CXX_STANDARD 11) # for constexpr specifier and other goodies
+set(BUILD_SHARED_LIBS ON)
```
Comment (lemire) [inline CMakeLists.txt:8]: If you force shared libraries (why?), you have to make sure that the system finds the shared library at runtime. Under Windows, you have to make sure that the DLL is found by the executables.

I see no reason to use a dynamic library. It makes the build more fragile. I recommend you leave it static.

Comment (KOLANICH) [reply CMakeLists.txt]: Though the CI still fails, likely because one of the tests takes too long.

**Comment by KOLANICH** [comment]
Hmm, it is definitely not timeout, but something else.

**Comment by lemire** [comment]
@KOLANICH The error indicates that the DLL is not found.

**Comment by lemire** [comment]
With a static build (the default in CMake), the library is included in the binaries. With a dynamic build, a distinct file (e.g., a DLL) is created in addition to the binary executable. In a portable way, one must then ensure that when the binary run, the dynamic library is found. This requires extra care.

**Comment by KOLANICH** [comment]
I'm sorry, IDK what is wrong. For me (Windows 7, 32-bit, MSYS2 MinGW-w64 + clang-12) everything builds (though there are issues with -lpthread not being added, I have tried to add it into PUBLIC dependencies of libfastpfor without much success (helped only for some libs), it helped to add it into all the exes, but I don't understand why I should do it.)

**Comment by lemire** [comment]
I am going to close this for now. We would love to have support for cpack and support dynamic libraries, but it is crucially important that we be able to continue running out tests successfully on various platforms. We cannot merge a PR that would break our build.

**Pull request closed without merging**